fairing/deployers/tfjob/tfjob.py

```diff
@@ -23,6 +23,31 @@ def generate_deployment_spec(self, pod_template_spec):
         """Returns a TFJob template"""
         self.set_container_name(pod_template_spec)
 
+        # Set appropriate secrets and volumes to enable kubeflow-user service
+        # account.
+        env_var = k8s_client.V1EnvVar(
```

Comment:
This will break in other platforms than gcp? cant this be optional?
same for volume_mount

Reply:
I agree, we'll need to add this as something configurable in the deployer

Reply:
we should probably have a global config that sets the deployment is happening in GCP and necessary things like gcloud for the builder and auth setup for the deployers are automatically configured.

Reply:
global config sets for deployer make sense. then we could also add other configurations like different custom labels for different jobs, volume, gpu setting etc.

Reply:
Right now tfjob superclasses job so maybe we just have an actual base deployer like we do for builders

Reply:
Right, thanks for the catch! In order to keep this PR manageable, I'll plan to add a flag to the TFJob deployer, and file a FR for the global config sets.

With the config sets it'll likely be useful to have a base deployer as well, in order for changes like these to be shared accordingly.

Reply:
Created #118